### PR TITLE
[4] Add Ensemble Methods

### PR DESCRIPTION
**Pull request by sofurihafe**
- [x] >10k dataset.
- [x] EDA.
- [x] **Bagging** и **AdaBoost** implementations.
- [x] Include implementations from **sklearn**: **Bagging**, **Random Forest**, **AdaBoost**, **Gradient Tree Boosting**, **Histogram-Based Gradient Boosting**.
- [x] **Hyperparameter optimization**, **cross-validation**.
- [x] Measure time, quality, accuracy.
- [x] * (**+2 points**) Include implementations from **XGBoost**, **LightGBM** и **CatBoost**.
- [x] * (**+3 points**) Use **Optuna**.

### CODE REVIEW

**Review by rustam-azimov — APPROVED** [review]
10 баллов
